File: idea/testData/hierarchy/calls/callees/javaMethod/main0.java
class KA {
    public KA() {

    }

    public final String name = "A";

    public final String foo(String s) {
        return "A " + s;
    }
}

class KClient {
    public final void <caret>bar() {
        new KA().foo("");
        new JA().foo(new JA().getName());

        new Runnable() {
            public void run() {
                new KA().foo("");
                new JA().foo(new JA().getName());
            }
        }.run();
    }
}

File: kotlin-ide/idea/testData/hierarchy/calls/callees/kotlinProperty/main0.kt
class KA {
    val name = "A"
    fun foo(s: String): String = "A: $s"
}

fun packageFun(s: String): String = s

val packageVal = ""

class KClient() {
    val <caret>bar: String
        get() {
            fun localFun(s: String): String = packageFun(s)
            val localVal = localFun("")

            KA().foo(KA().name)
            JA().foo(JA().name)
            localFun(packageVal)

            run {
                KA().foo(KA().name)
                JA().foo(JA().name)
                packageFun(localVal)
            }

            return ""
        }
}